experimental/dds/tree2/src/simple-tree/rawNode.ts

```diff
@@ -0,0 +1,276 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { FieldKey, TreeNodeSchemaIdentifier } from "../core";
+import {
+	EditableTreeEvents,
+	FieldNodeSchema,
+	FlexTreeContext,
+	FlexTreeField,
+	FlexTreeNode,
+	FlexTreeTypedNode,
+	FlexTreeEntityKind,
+	FlexTreeMapNode,
+	FlexTreeObjectNode,
+	FlexTreeTypedField,
+	FlexTreeUnboxField,
+	FlexibleFieldContent,
+	boxedIterator,
+	flexTreeMarker,
+	onNextChange,
+	LocalNodeKey,
+	MapNodeSchema,
+	ObjectNodeSchema,
+	TreeNodeSchema,
+	TreeNodeSchemaBase,
+	TreeStatus,
+	FlexTreeFieldNode,
+} from "../feature-libraries";
+import { fail } from "../util";
+import { InsertableTypedNode } from "./insertable";
+
+/** Stores the content of the raw node, i.e. the data that was passed to the factory */
+const nodeContent = Symbol();
+interface HasNodeContent<T> {
+	[nodeContent]: T;
+}
+
+/**
+ * Retrieve the content of a raw object node created via {@link createRawNode}.
+ * @remarks
+ * Raw nodes should only ever be processed as input once.
+ * Therefore, this will fail if called twice on the same node.
+ */
+export function extractRawNodeContent<
+	TSchema extends MapNodeSchema,
+	TContent extends InsertableTypedNode<TSchema>,
+>(node: RawMapNode<TSchema, TContent>): TContent;
+export function extractRawNodeContent<
+	TSchema extends FieldNodeSchema,
+	TContent extends InsertableTypedNode<TSchema>,
+>(node: RawFieldNode<TSchema, TContent>): TContent;
+export function extractRawNodeContent<TSchema extends ObjectNodeSchema, TContent extends object>(
+	node: RawObjectNode<TSchema, TContent>,
+): TContent;
+export function extractRawNodeContent(node: FlexTreeNode): object | undefined;
+export function extractRawNodeContent(node: object): object | undefined {
+	if (node instanceof RawTreeNode) {
+		const content = node[nodeContent] ?? fail("Node content may only be extracted once");
+		Reflect.deleteProperty(node, nodeContent);
+		return content as object;
+	}
+
+	return undefined;
+}
+
+/**
+ * Creates a node that falsely pretends to satisfy the given schema while wrapping the given node content.
```

Comment:
Nit (I think "pretends" is sufficient to convey the falsness): 
```suggestion
 * Creates a node that pretends to satisfy the given schema while wrapping the given node content.
```

Reply:
Good call.

Reply:
I supposed "falsely pretending" might actually imply that it's doing the real thing, correctly lol